client/landing/stepper/declarative-flow/readymade-template.tsx

```diff
@@ -0,0 +1,334 @@
+		const isLoggedIn = useSelector( isUserLoggedIn );
+		const currentUserSiteCount = useSelector( getCurrentUserSiteCount );
+		const currentPath = window.location.pathname;
+		const isCreateSite =
+			currentPath.endsWith( `setup/${ flowName }` ) ||
+			currentPath.endsWith( `setup/${ flowName }/` );
+
+		const userAlreadyHasSites = currentUserSiteCount && currentUserSiteCount > 0;
+
+		const locale = useFlowLocale();
+		const logInUrl = useLoginUrl( {
+			variationName: flowName,
+			redirectTo: window.location.href.replace( window.location.origin, '' ),
+			locale,
+		} );
+
+		useEffect( () => {
+			if ( ! isLoggedIn ) {
+				window.location.assign( logInUrl );
+			}
+		}, [] );
+
+		let result: AssertConditionResult = { state: AssertConditionState.SUCCESS };
+
+		if ( ! isLoggedIn ) {
+			result = {
+				state: AssertConditionState.CHECKING,
+				message: `${ flowName } requires a logged in user`,
+			};
+		} else if ( isCreateSite && ! userAlreadyHasSites ) {
+			result = {
+				state: AssertConditionState.CHECKING,
+				message: `${ flowName } with no preexisting sites`,
+			};
+		}
+
+		return result;
+	},
+};
+
+function enableAssemblerThemeAndConfigureTemplates(
+	themeId: string,
+	siteId: number,
+	readymadeTemplate: { content: string },
+	assembleSite: (
+		arg0: any,
+		arg1: string,
+		arg2: {
+			/**
+			 * @todo Separate the content in 3 sections. Ideally the entire template configuration should be done one the server, see the above comment.
+			 *
+			 * For now we piggyback on Site Assembler's API endpoint to apply the template on the site.
+			 */
+			homeHtml: any;
+			headerHtml: string;
+			footerHtml: string;
+			pages: never[];
+			globalStyles: object;
+			canReplaceContent: boolean;
+			// All sites using the assembler set the option wpcom_site_setup
+			siteSetupOption: string;
+		}
+	) => Promise< never >,
+	reduxDispatch: CalypsoDispatch
+) {
+	/**
+	 * This is not optimal. We should have one request for all of it that simply passes the readyMadeTemplateId.
+	 *
+	 * Right now we make 3 requests:
+	 * - Set the theme
+	 * - Get the Readymade template (with content)
+	 * - Apply the changes on the site (with the content that can be a large payload).
+	 */
+	return () =>
+		Promise.resolve()
+			.then( () =>
+				reduxDispatch(
+					activateOrInstallThenActivate( themeId, siteId, 'assembler', false ) as ThunkAction<
+						PromiseLike< string >,
+						any,
+						any,
+						AnyAction
+					>
+				)
+			)
+			.then( ( activeThemeStylesheet: string ) =>
+				assembleSite( siteId, activeThemeStylesheet, {
+					/**
+					 * @todo Separate the content in 3 sections. Ideally the entire template configuration should be done one the server, see the above comment.
+					 *
+					 * For now we piggyback on Site Assembler's API endpoint to apply the template on the site.
+					 */
+					homeHtml: readymadeTemplate.content,
+					headerHtml: '',
+					footerHtml: '',
+					pages: [],
+					globalStyles: {},
```

Comment:
this should apply the style variation that the selected design had active

Reply:
I created a separate issue for this: https://github.com/Automattic/dotcom-forge/issues/7944